Automattic/woocommerce-payments
# Update Platform Checkout Tracking conditions

#### Changes proposed in this Pull Request

With the recent changes in the usage tracking( p2: p2y3YZ-6L8-p2) , we can now collect usage data specific to WooPay regardless of the usage tracking opt-in status. This PR does the following so that we can receive `platform_checkout_enabled` and `platform_checkout_disabled` events regardless of usage tracking opt-in status.

- A minor refactor to change where additional event properties are being recorded.
- Introduce `maybe_record_admin_event` to record events emitted from the admin area
- Use `maybe_record_admin_event` to capture `platform_checkout_enabled` and `platform_checkout_disabled` events.








#### Testing instructions





Regression Test
- Make sure WooPay is enabled.
- Add a product to the cart and go to the checkout page.
- In ~5 minutes, check the Tracks Live View for the `woocommerceanalytics_order_checkout_start` event.
- Disable WooPay
- Add a product to the cart and go to the checkout page.
- `woocommerceanalytics_order_checkout_start` event should not appear in the Tracks Live view.

Check Admin events

- Optin to usage tracking in WooCommerce -> Settings -> Advanced -> WooCommerce.com -> Enable Tracking.
- Do the tests mentioned in this PR https://github.com/Automattic/woocommerce-payments/pull/3816. Note that setting `_wcpay_feature_platform_checkout` flag is no longer needed.
- Opt-out of tracking by disabling the checkbox set in step 1.
- Repeat the tests mentioned in https://github.com/Automattic/woocommerce-payments/pull/3816
- `wcadmin_platform_checkout_enabled` and `wcadmin_platform_checkout_disabled` events should still be registered.

Optional
- For good measure smoke test checkout with split upe enabled/disabled

-------------------


**Post merge**
